lib/ansible/cli/inventory.py

```diff
@@ -193,6 +197,14 @@ def dump(self, stuff):
             import yaml
             from ansible.parsing.yaml.dumper import AnsibleDumper
             results = yaml.dump(stuff, Dumper=AnsibleDumper, default_flow_style=False)
+        elif self.options.toml:
+            try:
+                from ansible.plugins.inventory.toml import toml_dumps
+            except ImportError:
```

Comment:
This won't happen unless you remove try/except from inventory module. However, if you want to keep it there, you could move `results = toml_dumps(stuff)` inside and catch `NameError` instead... (for example, there's other ways of course, like separating shims into a dedicated compat module)